File: Python/0747.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        largest = -1
        second_largest = -2
        answer = -1

        for i in range(len(nums)):
            if nums[i] >= largest:
                second_largest = largest
                largest = nums[i]
                answer = i
            elif nums[i] > second_largest and nums[i] < largest:
                second_largest = nums[i]

        print(second_largest)
        print(largest)
        print(answer)

        if largest < (second_largest * 2):
            return -1

        return answer

File: C/0747.c
int dominantIndex(int* nums, int numsSize) {
    int largest = -1, second_largest = -1, answer = -1;

    for (int i = 0; i < numsSize; i++) {
        if (nums[i] >= largest) {
            second_largest = largest;
            largest = nums[i];
            answer = i;
        } else if (nums[i] < largest && nums[i] > second_largest) {
            second_largest = nums[i];
        }
    }

    if (largest < (second_largest * 2)) {
        return -1;
    }

    return answer;
}
